src/libasr/codegen/asr_to_c_cpp.h

```diff
@@ -198,6 +199,27 @@ R"(#include <stdio.h>
         src = unit_src;
     }
 
+    void get_expr_src(ASR::expr_t *arg) {
```

Comment:
Can you document what this function does in a comment?

Reply:
When should one use it instead of `visit_expr` directly?

Reply:
Well this won't prevent direct usage of `visit_expr` in future. So we will have to keep collecting special cases again and add into this or another function. And the cycle will continue. 

Also, we won't know why some special case is handled here or which handling is for which node (thus making debugging difficult). Doing directly inside the node helps in figuring out the reasoning behind a particular check. 

IMO, such refactoring would make more sense in the long term future when LPython/LFortran will stabilise.

Reply:
What special case does it handle that has to be handled every time?

Reply:
If we come across `ListConstant`, `TupleConstant`, `DictConstant`, etc. which have some extra code-generation, we need to handle it in every node using if-else. So to avoid those repetitions in each node, I tried to extract these cases and wrap `visit_expr`.

Reply:
Then probably make it explicit with a function like `is_constant_aggregate_data_structure` and then call your new function. It should be explicit that why we are calling different function than `visit_expr`. But again not all nodes have been implemented yet so probably repeating code that handles special cases for now is okay. We will at least know what happens when we add a new constant and how existing node implementations are affected due to it.